FWCore/ParameterSet/python/Config.py

```diff
@@ -1824,6 +1874,62 @@ def apply(self,process):
                 self.__func(process)
                 self.__seenProcesses.add(process)
 
+class ProcessAccelerator(_ConfigureComponent,_Unlabelable):
+    """A class used to specify possible compute accelerators in a Process
+    instance. It is intended to be derived for any
+    accelerator/portability technology, and provides hooks such that a
+    specific customization can be applied to the Process on a worker
+    node at the point where the python configuration is serialized for C++.
+
+    The customizations must touch only untracked parameters. Each
+    deriving class should have a specific unit test enabling all
+    combinations of accelerators and assert that the configuration
+    hash does not change."""
+    def __init__(self):
+        pass
+    def _place(self, name, proc):
+        proc._placeAccelerator(self.type_(), self)
+    def type_(self):
+        return type(self).__name__
+    def dumpPython(self, options=PrintOptions()):
+        specialImportRegistry.registerUse(self)
+        result = self.__class__.__name__+"(" # not including cms. since the deriving classes are not in cms "namespace"
+        options.indent()
+        res = self.dumpPythonImpl(options)
+        options.unindent()
+        if len(res) > 0:
+            result += "\n"+res+"\n"
+        result += ")\n"
+        return result
+
```

Comment:
These extra spaces do not seem necessary

Reply:
I wanted to have a clear visual separation for the methods that are intended to be overridden by deriving classes, but perhaps one newline is sufficient.